sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventDataBatch.cs

```diff
@@ -180,34 +180,16 @@ public bool TryAdd(EventData eventData)
             {
                 AssertNotLocked();
 
-                var messageScopeCreated = false;
-                var traceparent = default(string);
+                ClientDiagnostics.InstrumentMessage(eventData.Properties, DiagnosticProperty.EventActivityName, out var traceparent, out var tracestate);
 
-                try
-                {
-                    ClientDiagnostics.InstrumentMessage(eventData.Properties, DiagnosticProperty.EventActivityName, out messageScopeCreated, out traceparent, out var tracestate);
-
-                    var added = InnerBatch.TryAdd(eventData);
-
-                    if ((added) && (traceparent != null))
-                    {
-                        EventDiagnosticIdentifiers.Add((traceparent, tracestate));
-                    }
+                var added = InnerBatch.TryAdd(eventData);
```

Comment:
If we do not accept the event into the batch, we should continue to remove the instrumentation, no?   Why would we keep it?